cache/cache.go

```diff
@@ -123,6 +137,37 @@ func (c *Cache[T]) Get(ctx context.Context, key string, fetcher Fetcher[T]) (val
 	return value, nil
 }
 
+// fill attempts to fetch a value from the upstream (using the passed fetcher)
+// and update the cache. It is called in the event of a hard cache miss.
+func (c *Cache[T]) fill(ctx context.Context, key string, fetcher Fetcher[T]) (value T, err error) {
+	log := logger.With(logging.GetFields(ctx)...).Sugar()
+
+	ctx, span := tracer.Start(
+		ctx,
+		"cache.miss",
+		trace.WithAttributes(c.spanAttributes(key)...),
+		trace.WithAttributes(attribute.String("cache.miss", "hard")),
+	)
+	defer span.End()
+
+	value, err = fetcher(ctx, key)
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+		return value, err
+	}
+
+	err = c.set(ctx, key, value)
+	if err != nil {
+		// Errors encountered while filling the cache are not returned to the
+		// caller: we don't want a cache availability problem to be exposed to the
+		// if the value was already successfully fetched.
```

Comment:
```suggestion
		// user if the value was already successfully fetched.
```

Guessing the word here! Maybe it should be `caller` instead?

Reply:
Thank you! I was trying not to repeat myself too much. Will take out "to the" instead.